util/types/json/serdes.go

```diff
@@ -305,11 +280,7 @@ func (m *jsonObject) decode(data []byte) (err error) {
 	return
 }
 
-func (a jsonArray) getTypeCode() byte {
-	return typeCodeArray
-}
-
-func (a jsonArray) encode(buffer *bytes.Buffer) {
+func encodeJSONArray(a []JSON, buffer *bytes.Buffer) {
```

Comment:
I do not think this is right place to do encoding. We should put encoding code to `util/codec` and add new type in `mysql` package. If you plan to do this in another PR, just ignore this comment.